### ISSUE TITLE
EIPIP Meeting 84

### ISSUE THREAD

**Issue by poojaranjan** [ethcatherders/EIPIP#245]
# Date and Time

June 28, 2023 at 14:00 UTC
# Location

Zoom: TBA in the Discord #eip-editing channel

YouTube Live Stream/Recording: https://www.youtube.com/playlist?list=PL4cwHXAawZxpLrRIkDlBjDUUrGgF91pQw

# Agenda
### 1. Discuss Open Issues/PRs, and other topics
- [x] https://github.com/ethereum/EIPs/issues/7198
- [x] https://github.com/ethereum/EIPs/pull/7222 
- [x] https://github.com/ethereum/EIPs/pull/7225 

#### Changes to `Final` proposals
_TBA_

### 2. Discussion continued or updates from past meetings
- [x] Discuss the EIP-ERC GitHub repo split Ref: [PR](https://github.com/ethereum/EIPs/pull/7206)
  - https://hackmd.io/A1zfRkO7RKKV6BIaSXpcbw
### 3. EIPs Insight - Monthly EIPs status reporting.
- [Hackmd](https://hackmd.io/@poojaranjan/EthereumImprovementProposalsInsight/https%3A%2F%2Fhackmd.io%2F%40poojaranjan%2FEIPsInsightJune2023)
- [EIPsInsight.com](https://eipsinsight.com/june-2023)

### 4. [EIP Editing Office Hour](https://hackmd.io/@poojaranjan/EIPEditingOfficeHour)
- Recording [Meeting 20](https://youtu.be/TX62kAjFkQg)
- Agenda [Meeting 21](https://github.com/ethereum-cat-herders/EIPIP/issues/246)

### 5. Review action items from earlier meetings
- Summary [here](https://github.com/ethereum-cat-herders/EIPIP/issues/241#issuecomment-1591487698) and [here](https://discord.com/channels/595666850260713488/746566142700814426/1118557698888323082)

**Comment by xinbenlv** [comment]
Propose to add to agenda: https://github.com/ethereum/EIPs/issues/7198

**Comment by timbeiko** [comment]
Given the support by client teams for https://github.com/ethereum/EIPs/pull/7206 on ACDE164, I'd like to discuss how to practically move the proposal forward.

**Comment by ulerdogan** [comment]
Propose to add the [EIP-7212](https://github.com/ethereum/EIPs/issues/7212) to the agenda.

**Comment by timbeiko** [comment]
@xinbenlv I think it probably still makes sense to discuss _how_ we'd approach this, and work through the "implementation details" on the call. If we want to wait to start implementing things until EIPIP85, so we can potentially incorporate feedback from ERC folks, I think that's reasonable. 

I would just recognize that the desire from the "core" side to split is pretty obvious at this point, so even if ERC authors don't "agree", they should probably think about "how to deal with the change" rather than "how to prevent it".

**Comment by poojaranjan** [comment]
@ulerdogan 
PR-7212 looks like a new proposal. Should you have any questions, perhaps that can be answered on [EIP Editing Office Hour 20](https://github.com/ethereum-cat-herders/EIPIP/issues/244).

**Comment by gcolvin** [comment]
https://github.com/ethereum/EIPs/pull/7206#issuecomment-1603684177

**Comment by gcolvin** [comment]
https://twitter.com/sproulM_/status/1671079363281051648?s=20

**Comment by gcolvin** [comment]
> > > Per ACD https://github.com/ethereum/EIPs/pull/164, we're going to move forward with this. We'll discuss more of the specifics on EIPIP next week.
> >
> > No, we are not. The Core Devs do not control the EIP organization and process.
> 
> I think this is an overly adversarial framing:

A PR was brought to the Core Devs on very short notice that had not even been discussed at an EIPIP meeting.  I'm not sure the Core Devs understood that this was an barely-reviewed, contentious PR -- not a consensus proposal by the Editors.  Then the claim was made that we are moving forward based on the Core Devs' consensus, rather than our own.  I object -- sometimes process really matters.

> Unfortunately, given the amount of discontent people generally feel about the process (you link to a Twitter example in your next comment), I think we're past the stage of "gathering feedback about what causes friction". There's a more general feeling that the EIP process is too rigid and high friction, as well as a concern that it cannot adapt to suit client devs' needs 

Agreed.

I've been clear all along that we need a larger plan that identifies and reduces the pain points and friction in our process.  Splitting the repos may or may not turn out to be part of that plan.

And yes, the "split the repos" debate has become a stand-in for larger political issues.  It has also seems to have highlighted  "philosophical" disagreements among the editors.   I'd really like to clean up our own house before engaging with the politics.  And yes, I think we know enough to not need more circular discussions.  We need plans on the table before can we make progress.

Towards that end, Victor and I have started a PR on EIP-1 that I'll add to the Agenda soon.  We've tried to identify and relieve some of the pain that we see, so that further discussion can focus on concrete proposals.  I do think that if we can fix the known problems in the process it will go a long way towards healing the political issues, and make it easier to further adapt our process to the needs of our users.

**Comment by xinbenlv** [comment]
As my commitment to my own principle of "I'll follow the consensus despite I might hold minority view", I put together a list of task items i think could be used as a starting point of discussing "How to split": [Task list for ERC/EIP Split](https://hackmd.io/A1zfRkO7RKKV6BIaSXpcbw). Please feel free to edit.

**Comment by abcoathup** [comment]
I support splitting ERCs from the EIP repo. The EIP/ERC processes should be given a chance to evolve for their users. 

Numbering whilst a trivial factor needs to be handled in the split. 
Existing ERCs shouldn't be renumbered. 
EIPs and ERCs shouldn't duplicate numbers. 
A shared number scheme is required. In the short term this could be a spreadsheet where the next sequential number is assigned by an editor. Though this still has issues of number gaming. Alternatively if editors have discretion then there could be a perception of preferential treatment to insiders. 
I'm happy to keep being involved in the numbering process. Whatever the process it should be quick at least for EIPs so they can easily be referred to.

Longer term a different number split could be looked at. Odds & evens (but ERCs vastly outnumber EIPs so there is either number wastage or ERC numbers would be out of sequence with EIP numbers.). Reserve a subset for EIPs e.g. 8xxx-9xxx. Or look at issuing the missing numbers. 

Sorry I can't attend the meeting because it isn't compatible with sleep in Australia.

**Comment by bumblefudge** [comment]
> Sorry I can't attend the meeting because it isn't compatible with sleep in Australia.

I've added your requirements to the [running hackmd doc](https://hackmd.io/A1zfRkO7RKKV6BIaSXpcbw?view), thanks for async contribution!

**Comment by poojaranjan** [comment]
## Summary 

### 1. Discuss Open Issues/PRs, and other topics
- Issue 7198: Responded with EIP-7199
- PR-7222: @xinbenlv mentioned it is blocking many changes and he is fine with whatever editors agree on.
- PR-7225: @lightclient @SamWilsn supports this. Will be moved to `Withdrawn`.

@xinbenlv will make a PR to EIP-1

### Bot Issue
- PR-7240 got approved to Final but not merged.
- @Samwlsn: waiting on updates on the bot situation from @Pandapip1 

### 2. Discussion continued or updates from past meetings
 #### Discuss the EIP-ERC GitHub repo split 
 - Juan pulled up the [Task list for ERC/EIP Split](https://hackmd.io/A1zfRkO7RKKV6BIaSXpcbw) hackmd and mentioned that he added some action points
 - Juan is willing to volunteer on the Editorial boards post-split 
 -  @gcolvin shared some process improvements https://github.com/ethereum/EIPs/pull/7230 
 - @xinbenlv suggested discussing points mentioned in Greg's PR (The PR is in "Draft")
 - @SamWilsn is open to discus Greg's proposal
 - Danno Ferrin brought CFI valid only for Core as an example of the requirement of Core process change 
 - The group discussed tooling. @SamWilsn will look into the EIPW bot and connect with @Pandapip1 for the eth-bot. Hopefully, both repos. will be managed by bots as it is now.
 -  @xinbenlv wants to collect feedback from ERC devs and share it by the next meeting.
 - @lightclient will hold merging the PR (7206) and wait till bots are in place
 - A lot of back and forth. It's worth watching the [recording](https://youtu.be/kgNGfCrOPfI).

The group will revisit progress in all directions and discuss merging PR-7206 in the [next meeting](https://github.com/ethereum-cat-herders/EIPIP/issues/248).

**Comment by poojaranjan** [comment]
Closing in favor of #248
